service/history/transferQueueStandbyProcessor.go

```diff
@@ -189,32 +212,67 @@ func (t *transferQueueStandbyProcessorImpl) processDecisionTask(transferTask *pe
 	sw := t.metricsClient.StartTimer(metrics.TransferStandbyTaskDecisionScope, metrics.TaskLatency)
 	defer sw.Stop()
 
+	var decisionScheduleToStartTimeout *int32
+	var tasklist *workflow.TaskList
 	processTaskIfClosed := false
+
+	execution := workflow.WorkflowExecution{
+		WorkflowId: common.StringPtr(transferTask.WorkflowID),
+		RunId:      common.StringPtr(transferTask.RunID),
+	}
+
 	return t.processTransfer(processTaskIfClosed, transferTask, func(msBuilder mutableState) error {
 		decisionInfo, isPending := msBuilder.GetPendingDecision(transferTask.ScheduleID)
 
+		executionInfo := msBuilder.GetExecutionInfo()
+		workflowTimeout := executionInfo.WorkflowTimeout
+		decisionTimeout := common.MinInt32(workflowTimeout, common.MaxTaskTimeout)
+		wfTypeName := executionInfo.WorkflowTypeName
+		startTimestamp := executionInfo.StartTimestamp
+
+		markWorkflowAsOpen := transferTask.ScheduleID <= common.FirstEventID+2 && msBuilder.IsWorkflowExecutionRunning()
+
 		if !isPending {
-			if transferTask.ScheduleID <= common.FirstEventID+2 {
-				return t.recordWorkflowStarted(msBuilder)
+			if markWorkflowAsOpen {
+				err := t.recordWorkflowStarted(transferTask.DomainID, execution, wfTypeName, startTimestamp.UnixNano(), workflowTimeout)
```

Comment:
Will this can called on every retry attempt?

Reply:
normally this should not be retried, unless the call to record workflow started encounters error, which will be (and should be) retried.